lib/Elevate/Components/MySQL.pm

```diff
@@ -110,19 +114,30 @@ sub _reinstall_mysql_packages {
             if ( $out =~ m{\sstate:\s*(\w+)} ) {
                 $status = $1;
             }
+
+            # we cannot trust the whmapi1 call (race condition) CPANEL-43253
+            if ( $status ne 'success' && --$wait_more > 0 ) {
+                sleep 1;
+                next;
+            }
+
             last;
         }
 
-        print "\n" if $c;    # clear the last "." from above
-        Cpanel::SafeRun::Simple::saferunnoerror( qw{/usr/local/cpanel/bin/whmapi1 configureservice service=mysql enabled=0} ) if !$enabled;
+        print "\n"                                                                                                          if $c;          # clear the last "." from above
+        Cpanel::SafeRun::Simple::saferunnoerror(qw{/usr/local/cpanel/bin/whmapi1 configureservice service=mysql enabled=0}) if !$enabled;
 
         if ( $status eq 'success' ) {
             INFO("MySQL $mysql_version restored");
         }
         else {
-            FATAL("Failed to restore MySQL $mysql_version: upgrade $id status '$status'");
+            my $msg = "Failed to restore MySQL $mysql_version: upgrade $id status '$status'";
+
+            FATAL($msg);
```

Comment:
does FATAL die?

Reply:
if it was the case that code would already be broken...

Reply:
`LOGDIE` is the one dying